lib/http2/stream.c

```diff
@@ -33,10 +33,13 @@ static size_t sz_min(size_t x, size_t y)
     return x < y ? x : y;
 }
 
+
+
 h2o_http2_stream_t *h2o_http2_stream_open(h2o_http2_conn_t *conn, uint32_t stream_id, h2o_req_t *src_req,
                                           const h2o_http2_priority_t *received_priority)
 {
-    h2o_http2_stream_t *stream = h2o_mem_alloc(sizeof(*stream));
+    h2o_http2_stream_t *stream = h2o_mem_alloc_recycle(&h2o_mem_req_allocator);
```

Comment:
Yes, we are going to benefit from these, I think.

It does not need to happen in this PR, but I think we want to do the same with HTTP/3 as well?